src/extension/webview.ts

```diff
@@ -0,0 +1,49 @@
+import {
+    Uri,
+    ViewColumn,
+    WebviewPanel,
+    window,
+    workspace,
+} from "vscode";
+
+export class PreviewPanel {
+    private pickedReport: string;
+    private previewPanel: WebviewPanel;
+
+    constructor(pickedReport: string) {
+        this.pickedReport = pickedReport;
+    }
+
+    public async createWebView() {
+        // Read in the report html and send it to the webview
+        const reportUri = Uri.file(this.pickedReport);
+        const reportHtml = await workspace.openTextDocument(reportUri);
+        const reportHtmlWithPolicy = this.addContentSecurityPolicy(reportHtml.getText());
+
+        // Construct the webview panel for the coverage report to live in
+        this.previewPanel = window.createWebviewPanel(
+            "coverageReportPreview",
+            "Preview Coverage Report",
+            ViewColumn.One,
+        );
+
+        this.previewPanel.webview.html = reportHtmlWithPolicy;
+    }
+
+    public dispose() {
+        this.previewPanel.dispose();
+    }
+
+    public addContentSecurityPolicy(text: string): string {
```

Comment:
this looks like it is only used by `PreviewPanel` maybe we should make this private?

Reply:
@ryanluker Thank you for the review. I initially set the `method` to be private, but the tests I wrote, require I make it public. Please I will need your advice on how to go about it

Reply:
I see! Lets leave it public then and merge this into the mainline.

Reply:
@ryanluker pleasure is mine... Thanks for being a patient guide

Reply:
@ryanluker waiting on this merge so I can take on a new issue

Reply:
Oh it does look like I need to merge this manually then... makes sense!